UI/Pages/TuitionPartner.cshtml.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const getSeoUrlFromId = (id) => id && id.replace("shortlist-tpInfo-cb-", "");
+const getAddToShortlistFetchUrl = () => `?handler=AddToShortlist`;
+const getRemoveFromShortlistFetchUrl = () => `?handler=RemoveFromShortlist`;
+const getAdditionalFetchParameter = (checkboxId) => ({
+  method: "POST",
+  body: JSON.stringify(`${getSeoUrlFromId(checkboxId)}`),
+  headers: {
+    RequestVerificationToken: document.getElementsByName(
+      "__RequestVerificationToken"
+    )[0].value,
+    "Content-Type": "application/json",
+    Accept: "application/json",
+  },
+});
+const getFetchResponseData = (response) => {
+  if (response.status >= 200 && response.status <= 299) {
+    return response.json();
+  } else {
+    throw Error(response.statusText);
+  }
+};
+const isResultValid = (result) =>
+  result && result.updated !== "undefined" && result.updated;
+const getError = (message = "Invalid result") => Error(message);
+const onChecked = async (checkboxId) => {
+  return await fetch(
+    getAddToShortlistFetchUrl(),
+    getAdditionalFetchParameter(checkboxId)
+  )
+    .then((response) => getFetchResponseData(response))
+    .catch((error) => {
+      throw Error(error);
+    });
+};
+const onUnChecked = async (checkboxId) => {
```

Comment:
As mentioned to you directly... should the "govuk-caption-l" be updated on check/uncheck to show the right caption like when the page loads?  TBC.